fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1289,24 +1294,47 @@ private boolean shouldClearExistingIndexEntries(@Nonnull IndexState indexState,
     }
 
     @Nonnull
-    private CompletableFuture<Void> doBuildIndexAsync(boolean markReadable) {
-        CompletableFuture<Void> buildFuture = buildEndpoints().thenCompose(tupleRange -> {
+    private CompletableFuture<Void> buildIndexFromRecordsAsync() {
+        return buildEndpoints().thenCompose(tupleRange -> {
             if (tupleRange != null) {
                 return buildRange(Key.Evaluated.fromTuple(tupleRange.getLow()), Key.Evaluated.fromTuple(tupleRange.getHigh()));
             } else {
                 return CompletableFuture.completedFuture(null);
             }
         });
+    }
+
+    @Nonnull
+    private CompletableFuture<Void> doBuildIndexAsync(boolean markReadable) {
+        boolean useIndexFromIndex = indexFromIndex != null && indexFromIndex.isActive();
```

Comment:
I think that having this nesting 'indexFromIndex' to hold a single value is an overkill. How about adding two fields to the builder: 
1. srcIndexForIndexScan - to build index from index 
2. allowFallbackToRecScan - this can be used for other scanning methods if and when